driver-core/src/main/com/mongodb/connection/ClusterDescription.java

```diff
@@ -97,6 +98,7 @@ public ClusterDescription(final ClusterConnectionMode connectionMode, final Clus
         this.serverDescriptions = new ArrayList<ServerDescription>(serverDescriptions);
         this.clusterSettings = clusterSettings;
         this.serverSettings = serverSettings;
+        this.logicalSessionTimeoutMinutes = calculateLogicalSessionTimeoutMinutes();
```

Comment:
Otherwise this is calculated over, and over, and over...

Reply:
I considered that, but new ClusterDescription instances are always created by the monitor threads so they don't affect the critical path.